File: Practice/src/exercise/SumOfDivisors.js
function solution(n) {
    return Array(n)
        .fill(1)
        .map((v, i) => v + i)
        .filter((i) => n % i == 0)
        .reduce((ac, c) => ac + c);

    // var answer = 0;
    // let i;
    // for (i = 1; i <= Math.sqrt(n); i++) {
    //     if (!(n % i)) {
    //         answer += i + n / i;
    //     }
    // }
    // i--;
    // return i === n / i ? answer - i : answer;

    // var answer = 0;
    // for (var i = 0; i < n + 1; i++) {
    //     if (n % i === 0) {
    //         answer += i;
    //     }
    // }
    // return answer;
}

File: Practice/src/exercise/SumOfDivisors.py
import math

def solution(n):
    answer = sum(i + n // i for i in range(1, int(math.sqrt(n)) + 1) if n % i == 0)
    return answer - (math.sqrt(n) if math.sqrt(n) == int(math.sqrt(n)) else 0)
    # sqrt = n ** 0.5
    # answer = sum(i + n // i for i in range(1, int(sqrt) + 1) if n % i == 0)
    # return answer - (sqrt if sqrt == int(sqrt) else 0)
